integration/kubernetes/helm-chart/alluxio/templates/proxy/daemonset.yaml

```diff
@@ -0,0 +1,99 @@
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+{{- if .Values.proxy.enabled }}
+{{- $hostNetwork := .Values.proxy.hostNetwork }}
+{{- $hostPID := .Values.proxy.hostPID }}
+{{- $fullName := include "alluxio.fullname" . }}
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ template "alluxio.fullname" . }}-proxy
+  labels:
+    app: {{ template "alluxio.name" . }}
+    chart: {{ template "alluxio.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: alluxio-proxy
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "alluxio.name" . }}
+      chart: {{ template "alluxio.chart" . }}
+      release: {{ .Release.Name }}
+      heritage: {{ .Release.Service }}
+      role: alluxio-proxy
+  template:
+    metadata:
+      labels:
+        app: {{ template "alluxio.name" . }}
+        chart: {{ template "alluxio.chart" . }}
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+        role: alluxio-proxy
+    spec:
+      hostNetwork: {{ $hostNetwork }}
+      hostPID: {{ $hostPID }}
+      dnsPolicy: {{ .Values.proxy.dnsPolicy | default ($hostNetwork | ternary "ClusterFirstWithHostNet" "ClusterFirst") }}
+      nodeSelector:
+      {{- if .Values.proxy.nodeSelector }}
+{{ toYaml .Values.proxy.nodeSelector | trim | indent 8  }}
+      {{- else if .Values.nodeSelector }}
+{{ toYaml .Values.nodeSelector | trim | indent 8  }}
+      {{- end }}
+      tolerations:
+      {{- if .Values.proxy.tolerations }}
+{{ toYaml .Values.proxy.tolerations | trim | indent 8  }}
+      {{- end }}
+      {{- if .Values.tolerations }}
+{{ toYaml .Values.tolerations | trim | indent 8  }}
+      {{- end }}
+      securityContext:
+        runAsUser: {{ .Values.user }}
+        runAsGroup: {{ .Values.group }}
+        fsGroup: {{ .Values.fsGroup }}
+    {{- if .Values.proxy.serviceAccount }}
+      serviceAccountName: {{ .Values.proxy.serviceAccount }}
+    {{- else if .Values.serviceAccount }}
+      serviceAccountName: {{ .Values.serviceAccount }}
+    {{- end }}
+      {{- if .Values.imagePullSecrets }}
+{{ include "alluxio.imagePullSecrets" . | indent 6 }}
+      {{- end}}
+      containers:
+        - name: alluxio-proxy
+          image: {{ .Values.image }}:{{ .Values.imageTag }}
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
+          {{- if .Values.proxy.resources  }}
+{{ include "alluxio.proxy.resources" . | indent 10 }}
+          {{- end }}
+          command: ["/entrypoint.sh"]
+          {{- if .Values.proxy.args }}
+          args:
+{{ toYaml .Values.proxy.args | trim | indent 12 }}
+          {{- end }}
+          env:
+          {{- range $key, $value := .Values.proxy.env }}
+          - name: "{{ $key }}"
+            value: "{{ $value }}"
+          {{- end }}
+          - name: ALLUXIO_CLIENT_HOSTNAME
+            valueFrom:
+              fieldRef:
+                fieldPath: status.hostIP
```

Comment:
should this be hostIP or containerIP?

Reply:
The worker fuse use hostIP, so I use hostIP here, but I'm not quite sure if we need this configuration? What do you think?

Reply:
I don't really know why FUSE needs hostIP, but I do know FUSE requires root permission to run so it has the permission to use hostIP. @ssz1997 Is hostIP a requirement for FUSE? I guess hostIP helps finding the local worker?

For Proxy, I believe hostIP should not be a requirement. The proxy is just for API translation and should only forward network traffic. @bzheng888 Could you test if proxy works without hostIP? Without looking at the proxy code I believe the answer is yes (and should be yes). Running the proxy should not need permission for hostIP.

Reply:
Thanks! Test on my environment the proxy works fine without hostIP, so I changed it to podIP, PTAL!

Reply:
Standalone Fuse is using hostNetwork (not sure why. Performance maybe) so there's no difference between podIP and hostIP. WorkerFuse that @bzheng888 was referring to should not be using hostIP when the worker is using podIP. However this `alluxio_client_hostname` property is not super important so didn't bring any trouble even if it's not correct.

Reply:
In the proxy case, as long as it's not using `hostNetwork`, it should use podIP.